### DISCUSSION TITLE
chapter_appendix/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#650] [Announcements]
# chapter_appendix/

动画图解、一键运行的数据结构与算法教程

https://www.hello-algo.com/chapter_appendix/

**Comment by zhgqcn** [comment]
大神，有没有兴趣补全这个网站呀。比如KMP算法等，你的教程真的是我学过最清晰易懂的！感谢有你

**Reply by krahets** [reply]
谢谢鼓励！这本书面向的是入门初学者，会优先写这部分内容，KMP 等后续有时间再写吧

**Reply by SeaEpoch** [reply]
@zhgqcn
KMP确实很难，总是不复习就会忘😢。
我写过一篇文章研究 KMP 的，希望对你有帮助！
[![banner](https://www.seayj.cn/articles/99ba/banner.jpg)](https://www.seayj.cn/articles/99ba/)

**Reply by Mycher7** [reply]
道友的博客网站那么好看😍

**Comment by windqwer** [comment]
请问后续可以加入一下最小生成树的相关内容吗，kruskal算法稍微有点难以理解，前面的图的遍历非常好，谢谢你！

**Reply by krahets** [reply]
这个近期不会添加，等有时间再写

**Reply by windqwer** [reply]
额，学习matlab有什么推荐的书籍或教程吗？非常感谢！

**Comment by gh9991** [comment]
感谢作者~~学习的过程很愉快*——*~！

**Reply by krahets** [reply]
祝贺读完，撒花🎉～

**Comment by JohnIreal** [comment]
感谢作者！

**Comment by smilelsb** [comment]
是否可以在附录这边添加关于语言的代码编写规范。
因为这里面的程序很多，如有有使用的代码编写规范，能让初学者更系统的知道，我设置这个变量，函数名代表什么意思。

**Reply by krahets** [reply]
Hi，书中的每种编程语言的代码风格通常都遵循了对应的主流 style guide 

> 我设置这个变量，函数名代表什么意思

这个是否代码注释已经包含了？

**Reply by smilelsb** [reply]
看书的过程已经发现代码风格遵循了对应的主流 style guide。
但是在太零散，可能讲到对应的地方才会出现注释，或者之前的注释有，不过后面再出现的就没有对应注释了，并且一直出现注释也不合适。
如果可以把不同语言所使用的风格做个统一说明，整合这些信息到附录中会让初学者更容易掌握编程规范，提高他们的代码质量和工作效率。
比如在python中，函数命名为什么要用下划线分开单词，常量为什么是全大写。如果都汇总在一起，有统一的规范，那将是极好的。

**Reply by smilelsb** [reply]
如果需要，您可以搭建好框架后，我可以尝试完成Python Style Guide，基于Google的。

**Reply by krahets** [reply]
我和 code reviewers 们讨论了一下，我们认为 code style guide 不太适合出现在本书中：

1. 代码注释的作用是描述代码含义，而非 code style ，所以不会出现你提到的 `可能讲到对应的地方才会出现注释，或者之前的注释有，不过后面再出现的就没有对应注释了`。
2. 虽然我们追求标准的代码风格，但这对理解算法关系不大。其更适合在学特定编程语言时去查阅了解。

**Reply by smilelsb** [reply]
了解~

**Comment by Vicshunyao** [comment]
点赞

**Comment by Matthew-CQ** [comment]
非常感谢作者们这么用心的成果，收获颇丰。我对于书中插图风格样式以及gif 动图印象颇深，因为一个明朗且美观的图解是非常不容易的。这里想请教一下作者作图工具是什么，以及风格样式当时是怎么参考的。谢谢！

**Comment by LilPear2002** [comment]
请问作者：
```text
Traceback (most recent call last):
  File "d:\pycharm_workspace\python_hello算法\chapter_tree\array_binary_tree.py", line 11, in <module>
    from modules import TreeNode, list_to_tree, print_tree
  File "d:\pycharm_workspace\python_hello算法\modules\__init__.py", line 11, in <module>
    from .tree_node import TreeNode, list_to_tree, tree_to_list
  File "d:\pycharm_workspace\python_hello算法\modules\tree_node.py", line 36, in <module>
    def list_to_tree_dfs(arr: list[int], i: int) -> TreeNode | None:
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

在 Python 3.10 及以后的版本中，可以使用 | 来表示类型联合，包括与 None 的联合。我的python版本是3.11 为什么还报错呢

**Comment by jiangnan2005** [comment]
完结撒花！！感谢所有的作者们

**Comment by xiaoyangst** [comment]
KMP 算法没有吗？

**Comment by Stellogic** [comment]
学完啦！！！感谢各位作者

**Comment by neko-inverse** [comment]
完结留念！看完一遍了，我还是要说，这绝对是算法学习史上一座丰碑一座里程碑，好比天帝开仙路，让凡人也能学习天阶功法，两个字，无敌！

**Comment by aftermathof** [comment]
看完了感谢，感觉要学的还有很多
